### DISCUSSION TITLE
General Discussion

### DISCUSSION THREAD

**Discussion by tyler-mairose-sp** [sailpoint-oss/sailpoint-cli#151] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by mpotti** [comment]
Hello I am wondering what is the best practices for switching between Sandbox and Production. Based on the documentation it seems that I would need to reset the environment variables each time I want to switch between them.

**Reply by tyler-mairose-sp** [reply]
Hey @mpotti,

If you are on a local machine where you interact with the cli daily I would recommend switching to PAT or oauth config.

There is a `sail env` command that will help you set this up.

You can run `sail env create <tenant_name>` and follow the steps to create that environment. Then you can run the same command for your prod environment.

Once both are configured you can select which to use by the following command `sail env use <environment_name>`.

You can run `sail env help` as well to see the other commands for managing environments.

**Comment by cpeckCC** [comment]
Hello,
  I noticed that the output download of transforms as the first 3 lines id, internal, name, type
  is at the end. And when we using the UI it is a the beginning.
  I am new SailPoint, is this can create issues or not ?

**Reply by mpotti** [reply]
This should not cause an issue. The reason that you are seeing them at the end is because the object which is being returned was setup that way.

**Comment by cpeckCC** [comment]
Thank you answering my question,
  Do you know if there is a command to download policies ?
   e.g.: sail policies list  and download

**Reply by cpeckCC** [reply]
More specifically it is called provision policy

**Comment by 192034576** [comment]
Looking for a little help with an install of CLI.  I'm trying to install on Linux and I'm getting these errors when trying to set the Client Secret for PAT:

`Error: exec: "dbus-launch": executable file not found in $PATH`

I then installed dbus and got past that error.  Now I get this error:

`Error: The name org.freedesktop.secrets was not provided by any .service files`

It looks like I may need to install a gnome keyring.  

I'm not sure I want to keep installing random components...  Any thoughts on this?  Do I simply need to keep following this trail of items that I need to install, one at a time?

**Reply by 192034576** [reply]
OK, it looks like the errors are related specifically to the storage of the client id/secret.  If I just create the environment without selecting an auth type and then supply the environment variables when I run the CLI, I don't get the errors and don't need to install any additional packages.
